config/initializers/semantic_logger.rb

```diff
@@ -52,3 +13,34 @@ def method_is_post_or_put_or_patch?
   )
   Rails.logger.info('Application logging to STDOUT')
 end
+
+## To avoid logging sensitive data on "subjects" and "to":
+## See more details on: https://github.com/reidmorrison/rails_semantic_logger/issues/230
+module RailsSemanticLogger
+  module ActionMailer
+    class LogSubscriber < ::ActiveSupport::LogSubscriber
+      class EventFormatter
+        def payload
+          filtered_params = Rails.application.config.filter_parameters
+          parameter_filter = ActiveSupport::ParameterFilter.new(filtered_params)
+
+          {}.tap do |h|
+            h[:event_name]         = event.name
+            h[:mailer]             = mailer
+            h[:action]             = action
+            h[:message_id]         = event.payload[:message_id]
+            h[:perform_deliveries] = event.payload[:perform_deliveries]
+            h[:subject]            = parameter_filter.filter(subject: event.payload[:subject])[:subject]
+            h[:to]                 = parameter_filter.filter(to: event.payload[:to])[:to]
+            h[:from]               = parameter_filter.filter(from: event.payload[:from])[:from]
+            h[:bcc]                = parameter_filter.filter(bcc: event.payload[:bcc])[:bcc]
+            h[:cc]                 = parameter_filter.filter(cc: event.payload[:cc])[:cc]
+            h[:date]               = date
+            h[:duration]           = event.duration.round(2) if log_duration?
+            h[:args]               = parameter_filter.filter(args: event.payload[:args])[:args]
```

Comment:
```suggestion
            h[:event_name]         = parameter_filter.filter_param('mailer.event_name', event.name)
            h[:mailer]             = parameter_filter.filter_param('mailer.mailer', mailer)
            h[:action]             = parameter_filter.filter_param('mailer.action', action)
            h[:message_id]         = parameter_filter.filter_param('mailer.message_id', event.payload[:message_id])
            h[:perform_deliveries] = parameter_filter.filter_param('mailer.perform_deliveries', event.payload[:perform_deliveries])
            h[:subject]            = parameter_filter.filter_param('mailer.subject', event.payload[:subject])
            h[:to]                 = parameter_filter.filter_param('mailer.to', event.payload[:to])
            h[:from]               = parameter_filter.filter_param('mailer.from', event.payload[:from])
            h[:bcc]                = parameter_filter.filter_param('mailer.bcc', event.payload[:bcc])
            h[:cc]                 = parameter_filter.filter_param('mailer.cc', event.payload[:cc])
            h[:date]               = parameter_filter.filter_param('mailer.date', date)
            h[:duration]           = parameter_filter.filter_param('mailer.duration', event.duration.round(2)) if log_duration?
            h[:args]               = parameter_filter.filter_param('mailer.args', event.payload[:args])
```

I had a slightly different approach here.

Unfortunately the tests don't cover this change 🤦‍♂️

Reply:
I also added 
```ruby
 'mailer.subject'
  'mailer.to'
  'mailer.args'
```
to https://github.com/DFE-Digital/apply-for-teacher-training/blob/54aa1e12ed9dad60cbe571d3cf053d3105350d7a/config/initializers/filter_parameter_logging.rb#L24-L26

Reply:
We could extract to a class to test this. I will do now